packages/synchronize/lib/src/journal.dart

```diff
@@ -0,0 +1,32 @@
+part of '../synchronize.dart';
+
+/// Contains the journal.
+///
+/// Used for detecting changes and new or deleted files.
+@JsonSerializable()
+class SyncJournal {
+  // Note: This must not be const as otherwise the entries are not modifiable when a const set is used!
+  // ignore: public_member_api_docs
+  SyncJournal(
+    this.entries,
+  );
+
+  // ignore: public_member_api_docs
+  factory SyncJournal.fromJson(final Map<String, dynamic> json) => _$SyncJournalFromJson(json);
+  // ignore: public_member_api_docs
+  Map<String, dynamic> toJson() => _$SyncJournalToJson(this);
+
+  /// All entries contained in the journal.
+  final Set<SyncJournalEntry> entries;
+
+  /// Update an [entry].
+  void updateEntry(final SyncJournalEntry entry) {
+    for (var i = 0; i < entries.length; i++) {
+      if (entries.elementAt(i) == entry) {
+        entries.remove(entries.elementAt(i));
+        break;
+      }
+    }
+    entries.add(entry);
```

Comment:
Isn't this the same as:

```suggestion
   entries..remove(entry)..add(entry);
```

Reply:
Yep :woman_facepalming: